### ISSUE TITLE
Application Crash when attempting to connect to `Ports > UInt16.max`

### ISSUE THREAD

**Issue by hishnash** [apple/swift-nio-transport-services#111]
### Expected behaviour

Throw `NIOTSErrors.InvalidPort` when a port is provided that is larger than `UInt16.max`

While having a port number larger than UInt16 should not happen. Given the publicly exposed api accepts Int and there is a `NIOTSErrors.InvalidPort` error exception users using Hypermedia style apis do find themselves at risk of a third party providing a (malformed) URL that crashes the application.

### Actual behaviour

Application crash. The Publicly exposed api expects.

### Steps to reproduce

1. Attempt to connect to a port number `> 65535` 

### If possible, minimal yet complete reproducer code (or URL to code)

Modify https://github.com/apple/swift-nio-transport-services/blob/main/Sources/NIOTSHTTPClient/main.swift#L70 replacing port value with `65536`

### SwiftNIO version/commit hash

`2.25.1`

### Swift & OS version (output of `swift --version && uname -a`)

```
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin20.3.0
Darwin Matthauss-MacBook-Pro-2.local 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64
```

**Comment by Lukasa** [comment]
Yes, I agree this is worth fixing. We can check it on the various bootstrap entry points.

**Comment by Lukasa** [comment]
Resolved by #112.
